etc/ci/performance/submit_to_perfherder.py

```diff
@@ -0,0 +1,380 @@
+        tj.add_coalesced_guid(data['job']['coalesced'])
+        tj.add_job_guid(data['job']['job_guid'])
+        tj.add_job_name(data['job']['name'])
+        tj.add_job_symbol(data['job']['job_symbol'])
+        tj.add_group_name(data['job']['group_name'])
+        tj.add_group_symbol(data['job']['group_symbol'])
+        tj.add_description(data['job']['desc'])
+        tj.add_product_name(data['job']['product_name'])
+        tj.add_state(data['job']['state'])
+        tj.add_result(data['job']['result'])
+        tj.add_reason(data['job']['reason'])
+        tj.add_who(data['job']['who'])
+        tj.add_tier(data['job']['tier'])
+        tj.add_submit_timestamp(data['job']['submit_timestamp'])
+        tj.add_start_timestamp(data['job']['start_timestamp'])
+        tj.add_end_timestamp(data['job']['end_timestamp'])
+        tj.add_machine(data['job']['machine'])
+
+        tj.add_build_info(
+            data['job']['build_platform']['os_name'],
+            data['job']['build_platform']['platform'],
+            data['job']['build_platform']['architecture']
+        )
+
+        tj.add_machine_info(
+            data['job']['machine_platform']['os_name'],
+            data['job']['machine_platform']['platform'],
+            data['job']['machine_platform']['architecture']
+        )
+
+        tj.add_option_collection(data['job']['option_collection'])
+
+        # for log_reference in data['job']['log_references']:
+        #    tj.add_log_reference( 'buildbot_text', log_reference['url'])
+
+        # data['artifact'] is a list of artifacts
+        for artifact_data in data['job']['artifacts']:
+            tj.add_artifact(
+                artifact_data['name'],
+                artifact_data['type'],
+                artifact_data['blob']
+            )
+        tjc.add(tj)
+
+        return tjc
+
+
+# TODO: refactor this big function to smaller chunks
+def submit(perf_data, failures, revision, summary, engine):
+
+    print("[DEBUG] failures:")
+    print(list(map(lambda x: x['testcase'], failures)))
+
+    author = "{} <{}>".format(revision['author']['name'],
+                              revision['author']['email'])
+
+    dataset = [
+        {
+            # The top-most revision in the list of commits for a push.
+            'revision': revision['commit'],
+            'author': author,
+            'push_timestamp': int(revision['author']['timestamp']),
+            'type': 'push',
+            # a list of revisions associated with the resultset. There should
+            # be at least one.
+            'revisions': [
+                {
+                    'comment': revision['subject'],
+                    'revision': revision['commit'],
+                    'repository': 'servo',
+                    'author': author
+                }
+            ]
+        }
+    ]
+
+    trsc = create_resultset_collection(dataset)
+
+    result = "success"
+    # FIXME: Always passing until https://bugzil.la/1276178 is fixed
```

Comment:
I remember Perfherder has other bad behavior on failed tests. I'll update the comment to reflect that.